va/va.h

```diff
@@ -1916,6 +2014,77 @@ typedef struct _VAContextParameterUpdateBuffer
     uint32_t reserved[VA_PADDING_MEDIUM];
 } VAContextParameterUpdateBuffer;
 
+/** \brief AES CTR encryption type */
+#define VA_ENCRYPTION_TYPE_CENC_CTR             0x00000001
+/** \brief AES CBC encryption type */
+#define VA_ENCRYPTION_TYPE_CENC_CBC             0x00000002
+
+/** \brief structure for encrypted segment info. */
+typedef struct _VAEncryptionSegmentInfo {
+  /** \brief  The offset relative to the start of the bitstream input in
+   *  bytes of the start of the segment */
+  uint32_t segment_start_offset;
+  /** \brief  The length of the segments in bytes */
+  uint32_t segment_length;
+  /** \brief  The length in bytes of the remainder of an incomplete block
+   *  from a previous segment*/
+  uint32_t partial_aes_block_size;
+  /** \brief  The length in bytes of the initial clear data */
+  uint32_t init_byte_length;
+  /** \brief  This will be AES counter for secure decode and secure encode
+   *  when numSegments equals 1, valid size is specified by
+   * \c key_blob_size */
+  uint8_t aes_cbc_iv_or_ctr[64];
+  /** \brief Reserved bytes for future use, must be zero */
+  uint32_t va_reserved[VA_PADDING_MEDIUM];
+} VAEncryptionSegmentInfo;
+
+/** \brief Encryption parameters buffer for VAEncryptionParameterBufferType */
+typedef struct _VAEncryptionParameters {
+  /** \brief Encryption type, refer to \c VA_ENCRYPTION_TYPE_CENC_CTR or
+   * \c VA_ENCRYPTION_TYPE_CENC_CBC */
+  uint32_t encryption_type;
+  /** \brief The number of sengments */
+  uint32_t num_segments;
+  /** \brief Pointer of segments */
+  VAEncryptionSegmentInfo *segment_info;
+  /** \brief The status report index for CENC workload.
+   *  The value is to indicate CENC workload and needs to be
+   *  different for each CENC workload */
+  uint32_t status_report_index;
```

Comment:
This parameter is included, which reference full sample decryption. However, there is no mention here of the other required structures for doing that such as VACencStatusParameters, VACencSliceParameterBufferH264 and  VACencStatusBuf.  Is this omission intentional and that will be carried with other headers? (i.e. https://source.chromium.org/chromium/chromium/src/+/master:third_party/libva_protected_content/va_protected_content.h)

Reply:
Definitely, the intention is to align comments with upstream files. Having said this, we can add specific comments for above structures, if these are no already present. Ryan, Could you add comments?